packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManager.java

```diff
@@ -630,7 +630,8 @@ private static Layout createLayout(
                 && maximumNumberOfLines != 0
                 && layout.getLineCount() > maximumNumberOfLines)
             || (heightYogaMeasureMode != YogaMeasureMode.UNDEFINED
-                && layout.getHeight() > height))) {
+                && layout.getHeight() > height)
+            || (text.length() == 1 && layout.getPaint().measureText(text.toString()) > width))) {
```

Comment:
This throws away metric effecting formatting on the text, like the font, whether its italic, how its justified, etc. It also throws out initial font size, which we only set on the paint after this.

If none of the existing line accessors in layout can tell that we've overflowed, and we really needed to do a new layout (we should avoid if possible), we would want to layout the full text, either using existing BoringLayout (simple not wrapping layout) which can give full width, or recreating a new layout without width constraint to get natural width. Though doing another layout here is really not ideal if there is any accessor on the existing layout that will give us this measurement.

Existing layout has [`fillCharacterBounds()`](https://developer.android.com/reference/android/text/Layout#fillCharacterBounds(int,%20int,%20float[],%20int)) accessor to get glyph rects in the existing layout. At a glance, if none of the line accessors show where the single glyph lies, this would I think let us avoid relayout.